src/mesh/serial_mesh.C

```diff
@@ -1271,6 +1271,18 @@ void ReplicatedMesh::stitching_helper (ReplicatedMesh * other_mesh,
               other_mesh->get_boundary_info().boundary_ids(other_elem, s, bc_ids);
               this->get_boundary_info().add_side(this_elem, s, bc_ids);
             }
+
+          // Copy shellface boundary ids
+          // This is only relevant for shell elements
+          if (other_elem->type()==TRISHELL3 || other_elem->type()==QUADSHELL4)
```

Comment:
Agreed.  Even if we add is_shellface() and we only want to allow shellface boundary ids there, the place to enforce that is in the BoundaryInfo methods which add shellface ids, not in every code that calls those methods.